### PR TITLE
[ALREADY MERGED IN 0.8.6] Feature/migration to HTTP transport + message bufferization when admin is unavailable 10741

### PR DESCRIPTION
**Pull request by epamrsa**

### CODE REVIEW

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

**Comment by RomanDavlyatshin** [comment]
Converting to draft bc `feature/implement-java-agent-messages-bufferization-during-admin-unavailabiity-EPMDJ-107411` is already merged in #138 
This PR is kept around to preserve TODO comments

**Comment by RomanDavlyatshin** [comment]
already merged at https://github.com/Drill4J/java-agent/pull/138

File: test2code/src/main/kotlin/com/epam/drill/test2code/Test2Code.kt [hunk 1]
```diff
@@ -129,17 +111,25 @@ class Test2Code(
...
-    private fun sendMessage(message: CoverMessage) {
-        val messageStr = json.encodeToString(CoverMessage.serializer(), message)
-        logger.debug { "Send message $messageStr" }
-        send(messageStr)
+    private val classMetadataDestination = AgentMessageDestination("POST", "class-metadata")
+    private val classMetadataCompleteDestination = AgentMessageDestination("POST", "class-metadata/complete")
+
+    private fun sendClassMetadata(astEntities: List<AstEntity>) {
+        val message = ClassMetadata(astEntities = astEntities)
+        logger.debug { "sendClassMetadata: Sending class metadata: $message" }
```
[the remark this answers is not in the document]
Comment (epamrsa) [reply test2code/src/main/kotlin/com/epam/drill/test2code/Test2Code.kt:126]: I think in this case dubug is optimal, as this event occurred only once per agent launch. TRACE is typically used for frequently repeated messages, like send coverage data.

File: test2code/src/main/kotlin/com/epam/drill/test2code/Test2Code.kt [hunk 2]
```diff
@@ -129,17 +111,25 @@ class Test2Code(
...
+        sender.send(classMetadataDestination, message)
+    }
+
+    private fun sendClassMetadataComplete() {
+        logger.debug { "sendClassMetadataComplete: Sending class metadata complete message" }
```
[the remark this answers is not in the document]
Comment (epamrsa) [reply test2code/src/main/kotlin/com/epam/drill/test2code/Test2Code.kt:131]: I think in this case dubug is optimal, as this event occurred only once per agent launch. TRACE is typically used for frequently repeated messages, like send coverage data.

File: test2code/src/main/kotlin/com/epam/drill/test2code/Test2Code.kt [hunk 3]
```diff
@@ -129,17 +111,25 @@ class Test2Code(
         scanClasses { classes ->
             classes
                 .map { parseAstClass(it.entityName(), it.bytes()) }
-                .let(::InitDataPart)
-                .also(::sendMessage)
-                .also { classCount += it.astEntities.size }
+                .also(::sendClassMetadata)
```
[the remark this answers is not in the document]
Comment (epamrsa) [reply test2code/src/main/kotlin/com/epam/drill/test2code/Test2Code.kt:114]: Just matter of code style, the pair of sendClassMetadata and sendClassMetadataComplete methods looks logical and understandable.

**Pull request closed without merging**